geopandas/tests/test_sindex.py

```diff
@@ -172,10 +174,284 @@ def test_intersection_invalid_bounds_tuple(self, test_geom):
+            (Point(), []),
+            (MultiPolygon(), []),
+            (Polygon(), []),
+        ],
+    )
+    def test_query_empty_geometry(self, test_geom, expected_value):
+        """Tests the `query` method with empty geometry.
+        """
+        res = self.df.sindex.query(test_geom)
+        assert_array_equal(res, expected_value)
+
+    @pytest.mark.parametrize(
+        "predicate, test_geom, expected", (("test", (-1, -1, -0.5, -0.5), [[], []]),),
+    )
+    def test_query_invalid_predicate(self, predicate, test_geom, expected):
+        """Tests the `query` method with invalid predicates.
+        """
+        test_geom = box(*test_geom)
+        with pytest.raises(ValueError):
+            self.df.sindex.query(test_geom, predicate=predicate)
+
+    @pytest.mark.parametrize(
+        "sort, expected",
+        (
+            (True, [[0, 0, 0], [0, 1, 2]]),
+            # False could be anything, at least we'll know if it changes
+            (False, [[0, 0, 0], [0, 1, 2]]),
+        ),
+    )
+    def test_query_sorting(self, sort, expected):
+        """Check that results from `query` don't depend on the
+        order of geometries.
+        """
+        # these geometries come from a reported issue:
+        # https://github.com/geopandas/geopandas/issues/1337
+        # there is no theoretical reason they were chosen
+        test_polys = GeoSeries([Polygon([(1, 1), (3, 1), (3, 3), (1, 3)])])
+        tree_polys = GeoSeries(
+            [
+                Polygon([(1, 1), (3, 1), (3, 3), (1, 3)]),
+                Polygon([(-1, 1), (1, 1), (1, 3), (-1, 3)]),
+                Polygon([(3, 3), (5, 3), (5, 5), (3, 5)]),
+            ]
+        )
+        expected = [0, 1, 2]
+
+        # pass through GeoSeries to have GeoPandas
+        # determine if it should use shapely or pygeos geometry objects
+        tree_df = geopandas.GeoDataFrame(geometry=tree_polys)
+        test_df = geopandas.GeoDataFrame(geometry=test_polys)
+
+        test_geo = test_df.geometry.values.data[0]
+        res = tree_df.sindex.query(test_geo, sort=sort)
+        try:
+            assert_array_equal(res, expected)
+        except AssertionError as e:
+            if not compat.USE_PYGEOS and sort is False:
+                pytest.xfail(
+                    "rtree results are known to be unordered, see "
+                    "https://github.com/geopandas/geopandas/issues/1337"
+                )
+            raise e
+
+    # ------------------------- `query_bulk` tests -------------------------- #
+    @pytest.mark.parametrize(
+        "predicate, test_geom, expected",
+        (
+            (None, (-1, -1, -0.5, -0.5), [[], []]),
+            (None, (-0.5, -0.5, 0.5, 0.5), [[0], [0]]),
+            (None, (0, 0, 1, 1), [[0, 0], [0, 1]]),
+            ("intersects", (-1, -1, -0.5, -0.5), [[], []]),
+            ("intersects", (-0.5, -0.5, 0.5, 0.5), [[0], [0]]),
+            ("intersects", (0, 0, 1, 1), [[0, 0], [0, 1]]),
```

Comment:
Also add test cases for multiple input geometries; ideally 2 variants:
- both geometries intersect
- only second geometry interesects

Reply:
Added these two types of tests.